index.d.ts

```diff
@@ -66,6 +66,8 @@ declare namespace fastifyOauth2 {
     generateStateFunction?: Function;
     checkStateFunction?: Function;
     startRedirectPath: string;
+    tags?: string[];
+    schema?: Object;
```

Comment:
```suggestion
    schema?: object;
```
we want only non primitve values here: https://2ality.com/2020/01/typing-objects-typescript.html#object-vs.-object%3A-primitive-values

Even better solution should be using schema typings from  https://www.npmjs.com/package/@types/swagger-schema-official as we are already doing in [`fastify-swagger` plugin](https://github.com/fastify/fastify-swagger/blob/master/index.d.ts#L2): the interface is called simply [`Schema`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/swagger-schema-official/index.d.ts#L176)

Reply:
I will use ```object``` for the test, but ```FastifySchema``` as the typescript typing.
The main reason is that the schema option is directly passed to fastify route options.

Reply:
schema should be more like [```FastifySchema```](https://github.com/fastify/fastify-swagger/blob/master/index.d.ts#L13) instead of [```Schema```](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/swagger-schema-official/index.d.ts#L176). It is because I managed to pass the type directly to the route options. Schema of ```@types/swagger-schema-official``` is missing those extra parameters.

Therefore, I manage to use ```object``` to reduce dependence.